packages/electron/src/fetchParity.ts

```diff
@@ -0,0 +1,187 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: MIT
+
+import { app, BrowserWindow } from 'electron';
+import axios from 'axios';
+import { file } from 'checksum';
+import { download } from 'electron-dl';
+import { chmod, rename, unlink } from 'fs';
+import { promisify } from 'util';
+import * as retry from 'async-retry';
+
+import { defaultParityPath, getParityPath } from './getParityPath';
+import logger from './utils/logger';
+
+interface FetchParityOptions {
+  onProgress: (progress: number) => void;
+  parityChannel: string;
+}
+
+/**
+ * @ignore
+ */
+const checksum = promisify(file);
+/**
+ * @ignore
+ */
+const fsChmod = promisify(chmod);
+/**
+ * @ignore
+ */
+const fsRename = promisify(rename);
+/**
+ * @ignore
+ */
+const fsUnlink = promisify(unlink);
+
+/**
+ * @ignore
+ */
+const VANITY_URL = 'https://vanity-service.parity.io/parity-binaries';
+
+/**
+ * @ignore
+ */
+const getArch = () => {
+  switch (process.platform) {
+    case 'darwin':
+    case 'win32':
+      return 'x86_64';
+    default: {
+      switch (process.arch) {
+        case 'arm':
+          return 'arm';
+        case 'arm64':
+          return 'aarch64';
+        case 'x32':
+          return 'i686';
+        default:
+          return 'x86_64';
+      }
+    }
+  }
+};
+
+/**
+ * @ignore
+ */
+const getOs = () => {
+  switch (process.platform) {
+    case 'darwin':
+      return 'darwin';
+    case 'win32':
+      return 'windows';
+    default:
+      return 'linux';
+  }
+};
+
+/**
+ * Remove parity binary or partial binary in the userData folder, if it exists.
+ */
+export async function deleteParity() {
+  const parityPath = await defaultParityPath();
+
+  // Remove parity binary
+  try {
+    await fsUnlink(parityPath);
+  } catch (e) {
+    /* Do nothing if error. */
+  }
+
+  // Remove parity partial binary (download was still in progress)
+  try {
+    await fsUnlink(`${parityPath}.part`);
+  } catch (e) {
+    /* Do nothing if error. */
+  }
+}
+
+/**
+ * Downloads Parity, saves it to Electron's `userData` folder, and returns the
+ * path to the downloaded binary once finished.
+ */
+export async function fetchParity(
+  mainWindow: BrowserWindow,
+  options: FetchParityOptions = {
```

Comment:
ditto cf https://github.com/paritytech/fether/pull/163/commits/752ec5f4fc7172c84d2dc4fe1defe1d5f07d2f9f
I think if we just call `fetchParity({parityChannel: 'stable'})` it would crash because `onProgress` will be undefined and the function will try to call it